backend/tessend/tessend.go

```diff
@@ -123,7 +149,41 @@ func (t *Tessend) Err() <-chan error {
 
 // Name returns the daemon name.
 func (t *Tessend) Name() string {
-	return "tessend"
+	return componentName
+}
+
+// Receiver returns the tessen receiver channel.
+func (t *Tessend) Receiver() chan<- interface{} {
+	return t.messageChan
+}
+
+func (t *Tessend) startMessageHandler() {
+	for {
+		msg, ok := <-t.messageChan
+		if !ok {
```

Comment:
This will only happen when the daemon is stopped, and since the closing happens in the same package I don't really think this is an error condition. I think it's probably fine to just return here.